src/common/components/repository-input/index.js

```diff
@@ -33,6 +34,14 @@ export class RepositoryInput extends Component {
     return message;
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
initial suggestion here http://jamesknelson.com/join-the-dark-side-of-the-flux-responding-to-actions-with-actors/ is to do it in connect? then goes on to talk about actors ...